contracts/multisigs/PolySafeCreatorWithRecoveryModule.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.30;
+
+// Poly Safe Proxy Factory: https://polygonscan.com/address/0xaacfeea03eb1561c4e67d661e40682bd20e3541b#code
+interface IPolySafeProxyFactory {
+    // Signature struct
+    struct Sig {
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+
+    /// @dev Creates Poly Safe proxy contract.
+    function createProxy(address paymentToken, uint256 payment, address payable paymentReceiver, Sig memory createSig)
+        external;
+
+    /// @dev Computes Poly Safe proxy address based on owner account.
+    function computeProxyAddress(address owner) external view returns (address);
+}
+
+// Safe Master Copy corresponding to Polygon mainnet: https://polygonscan.com/address/0xd9db270c1b5e3bd161e8c8503c55ceabee709552#code
+interface ISafe {
+    /// @dev Gets Safe's set of owners.
+    function getOwners() external view returns (address[] memory);
+
+    /// @dev Gets Safe threshold.
+    function getThreshold() external view returns (uint256);
+}
+
+/// @dev Provided zero address.
+error ZeroAddress();
+
+/// @dev Multisig already exists.
+/// @param multisig Multisig address.
+error MultisigAlreadyExists(address multisig);
+
+/// @dev Provided incorrect multisig threshold.
+/// @param expected Expected threshold.
+/// @param provided Provided threshold.
+error WrongThreshold(uint256 expected, uint256 provided);
+
+/// @dev Provided incorrect number of owners.
+/// @param expected Expected number of owners.
+/// @param provided Provided number of owners.
+error WrongNumOwners(uint256 expected, uint256 provided);
+
+/// @dev Provided incorrect multisig owner.
+/// @param provided Provided owner address.
+error WrongOwner(address provided);
+
+/// @title PolySafeCreatorWithRecoveryModule - Smart contract for Poly Safe multisig implementation with Recovery Module
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+/// @author Mariapia Moscatiello - <mariapia.moscatiello@valory.xyz>
+contract PolySafeCreatorWithRecoveryModule {
+    event MultisigCreated(address indexed multisig, address indexed owner);
```

Comment:
I think we should keep just multisig to be compatible with subgraph. But worth to check that with Axat